### PR TITLE
Update account header to match Figma

### LINKED ISSUE #1945: Account header on Dashboard and Log should be updated to match Figma
The account header has been redesigned, and some of our assumptions moving forward rely on this layout.

Some requirements:

[ ] The "Create Lock" button has 2-pronged behavior
  [ ] On the dashboard, it opens the Create Lock form
  [ ] On the log page, it moves you to the Dashboard (and also opens the form?)

[ ] The search will have to be implemented
  [ ] Initial version can be fairly naive
  [ ] Later versions will likely involve some interaction with Locksmith.

Complexity estimates:
 - Header Layout: simple
 - Button Behavior: medium difficult
 - Search: complex

### PR DESCRIPTION
**Pull request by cnasc**
# Description

This PR is a huge sweeping change to bring our UI back into sync with the Figma. It lifts the create lock button out of the list of locks and into the omnipresent account header within `unlock-app`.

I will keep this as a draft for the moment.

image

# Issues

Fixes #1945 

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [x] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [x] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [x] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/theme/media.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import styled, { css } from 'styled-components'
```
Comment (cnasc) [inline unlock-app/src/theme/media.ts]: Changing this file was a huge but necessary pain, thanks to TypeScript. Note that the types are not very concrete; at some point in the future this should be rectified.

Comment (julien51) [reply unlock-app/src/theme/media.ts]: ok, that should make for easy splitting!

File: unlock-app/src/components/content/DashboardContent.js [hunk 2]
```diff
@@ -35,35 +59,54 @@ DashboardContent.propTypes = {
...
+    network,
     lockFeed,
+    formIsVisible,
   }
 }
 
-export default connect(mapStateToProps)(DashboardContent)
+export default connect(
+  mapStateToProps,
+  actionCreators
```
Comment (julien51) [inline unlock-app/src/components/content/DashboardContent.js]: Please use `mapDispatchToProps` for consistency with the rest of the app.

**Comment by cnasc** [comment]
Closing in favor of #2369

**Pull request closed without merging**